File: oob-webapp.disabled/webapp/src/api/board.ts
import axios from "axios";
import { Board, BoardHostname } from "../entities";

export async function readBoard(): Promise<Board> {
  const { data } = await axios.get<Board>("/api/board");

  return data;
}

export async function readHostname(): Promise<BoardHostname> {
  const { data } = await axios.get<BoardHostname>("/api/board/hostname");

  return data;
}

export async function updateHostname(params: {
  hostname: string;
}): Promise<BoardHostname> {
  const { hostname } = params;

  const { data } = await axios.put("/api/board/hostname", {
    hostname,
  });

  return data;
}

File: arduino-ootb.disabled/go-webapp-arduino/api/routes/board.go
package routes

import (
	"fmt"
	"net/http"
	"x8-oob/board"

	"github.com/labstack/echo/v4"
)

type boardInfo struct {
	DeviceModelName string `json:"deviceModelName"`
}

type updateHostnameBody struct {
	Hostname string `json:"hostname"`
}

type boardHostname struct {
	Hostname string `json:"hostname"`
}

func ReadBoard(c echo.Context) error {
	b := boardInfo{
		DeviceModelName: "Arduino Portenta X8",
	}

	return c.JSON(http.StatusOK, b)
}

func ReadHostname(c echo.Context) error {
	h, err := board.ReadHostname()
	if err != nil {
		return c.JSON(http.StatusInternalServerError, err)
	}

	b := boardHostname{
		Hostname: h,
	}

	return c.JSON(http.StatusOK, b)
}

func UpdateHostname(c echo.Context) error {
	b := updateHostnameBody{}

	err := c.Bind(&b)
	if err != nil {
		return c.JSON(http.StatusBadRequest, fmt.Errorf("parsing body: %w", err))
	}

	fmt.Println(b.Hostname)

	h, err := board.UpdateHostname(b.Hostname)
	if err != nil {
		return c.JSON(http.StatusInternalServerError, err)
	}

	r := boardHostname{
		Hostname: h,
	}

	return c.JSON(http.StatusOK, r)
}
